cloudsync/youtube_test.py

```diff
@@ -0,0 +1,185 @@
+    with pytest.raises(Exception) as exc:
+        YouTubeApi().upload_video(videofile.video)
+    assert str(exc.value).startswith('Retried YouTube upload 10x') == retryable
+
+
+def test_upload_caption_calls_insert(mocker):
+    """
+    Test that the upload_caption task calls insert_caption for a YouTube video if no caption for that language exists
+    """
+    subtitle = VideoSubtitleFactory()
+    caption_id = 'foo'
+    caption_response = {'id': caption_id}
+    mocker.patch('cloudsync.youtube.YouTubeApi.list_captions', return_value={})
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().captions.return_value.insert.return_value.next_chunk.return_value = (None, caption_response)
+    response = YouTubeApi().upload_caption(subtitle, caption_id)
+    assert response == caption_response
+
+
+def test_upload_caption_calls_update(mocker):
+    """
+    Test that the upload_caption task calls update_caption for a YouTube video if a caption for that language exists
+    """
+    subtitle = VideoSubtitleFactory()
+    caption_id = 'bar'
+    caption_response = {'id': caption_id}
+    mocker.patch('cloudsync.youtube.YouTubeApi.list_captions', return_value={'en': caption_id})
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().captions.return_value.update.return_value.next_chunk.return_value = (None, caption_response)
+    response = YouTubeApi().upload_caption(subtitle, caption_id)
+    assert response == caption_response
+
+
+def test_delete_video(mocker):
+    """
+    Test that the 'delete_video' method executes a YouTube API deletion request and returns the status code
+    """
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().videos.return_value.delete.return_value.execute.return_value = 204
+    assert YouTubeApi().delete_video('foo') == 204
+    youtube_mocker().videos.return_value.delete.assert_called_with(id='foo')
+
+
+def test_list_captions(mocker):
+    """
+    Test that the 'list_captions' method executes a YouTube API request and returns a dict with correct values
+    """
+    key = 'en'
+    value = 'Srnr982VEC79QzEBGcBOL_UFmu9U2e-JgOw-EWIxJXEB5Bjltl3Yvg='
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().captions.return_value.list.return_value.execute.return_value = {
+        'etag': 'foo',
+        'items': [{
+            'id': value,
+            'kind': 'youtube#caption',
+            'snippet': {
+                'audioTrackType': 'unknown',
+                'language': key,
+                'lastUpdated': '2017-11-15T14:53:21.839Z',
+                'name': 'English',
+                'videoId': '3h-0mkTVbRg'
+            }
+        }],
+        'kind': 'youtube#captionListResponse'
+    }
+    assert YouTubeApi().list_captions('foo') == {key: value}
+    assert youtube_mocker().captions.return_value.list.called_with('foo')
```

Comment:
You need to use `assert_called_with` or a related function here. Since mocked objects are truthy assertions like `assert x().y().z()` will always evaluate to true.

Reply:
changed